### PR TITLE
Structurally normalize weak and inherent in new solver

### PR DESCRIPTION
**Pull request by compiler-errors**
It seems pretty obvious to me that we should be normalizing weak and inherent aliases too, since they can always be normalized. This PR still leaves open the question of what to do with opaques, though 💀

**Also**, we need to structurally resolve the target of a coercion, for the UI test to work.

r? @lcnr

### CODE REVIEW

**Comment by compiler-errors** [comment]
I thought I already had this included somewhere in some other PR, but I may have rebased it out by accident. Maybe I totally forgor and this is already up in some other PR.

File: compiler/rustc_hir_typeck/src/coercion.rs [hunk 1]
```diff
@@ -1012,6 +1012,10 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         cause: Option<ObligationCause<'tcx>>,
     ) -> RelateResult<'tcx, Ty<'tcx>> {
         let source = self.try_structurally_resolve_type(expr.span, expr_ty);
+        let target = self.try_structurally_resolve_type(
```
Comment (compiler-errors) [inline compiler/rustc_hir_typeck/src/coercion.rs:1015]: I take it back -- @lqd: this line may have changed the order we process obligations, re: perf #114604. 

I'd be surprised if this had a meaningful effect on perf, tho.

Comment (lqd) [reply compiler/rustc_hir_typeck/src/coercion.rs:1015]: I’ll check if cachegrind has anything interesting to say

Comment (compiler-errors) [reply compiler/rustc_hir_typeck/src/coercion.rs:1015]: I'm also just getting rid of this one line and running perf in #114648.

Comment (compiler-errors) [reply compiler/rustc_hir_typeck/src/coercion.rs:1015]: If this is the culprit, I guess I may be able to work around it.

**Pull request merged**